### PR TITLE
CORTX-32737: Changes related to fetching configs from consul  gconf

### PR DESCRIPTION
**Pull request by rohitkolapkar**
Signed-off-by: Rohit Kolapkar <rohit.j.kolapkar@seagate.com>

# Pull Request
## Problem Statement
-   Problem statement

## Design
-   For Bug, Describe the fix here.
-   For Feature, Post the link for design

-   \[ \] Coding conventions are followed and code is consistent

-   \[ \] Unit and System Tests are added
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[ \] Testing was performed with RPM

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[ \] JIRA number/GitHub Issue added to PR
-   \[ \] PR is self reviewed
-   \[ \] Jira and state/status is updated and JIRA is updated with PR link
-   \[ \] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

## Dev Testing

1. Login working --  indicates User management working
image

2. IAM user creation working fine
image

3. bytecount status
image

4. Deployment using provisioner's gconf branch - mini provisioning passed
image

5. Deployment using provisioner's gconf branch - control pod up and running, data pods failure expected.
image

### CODE REVIEW

File: csm/conf/configure.py [hunk 1]
```diff
@@ -156,16 +153,16 @@ def _set_csm_endpoint():
 
     @staticmethod
     def _set_s3_endpoints():
-        s3_endpoints = Conf.get(const.CONSUMER_INDEX, const.RGW_S3_DATA_ENDPOINTS_KEY)
-        if s3_endpoints:
-            Log.info(f"Fetching s3 endpoint.{s3_endpoints}")
-            s3_endpoints_count = len(s3_endpoints)
-            for endpoint_count in range(s3_endpoints_count):
+        s3_num_eps = Conf.get(const.CONSUMER_INDEX,
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:156]: s3_num_eps -> count_endpoints

File: csm/conf/configure.py [hunk 2]
```diff
@@ -156,16 +153,16 @@ def _set_csm_endpoint():
...
+            const.RGW_NUM_ENDPOINTS_KEY)
+        for ep_count in range(int(s3_num_eps)):
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:158]: ep_count -> count

File: csm/conf/configure.py [hunk 3]
```diff
@@ -156,16 +153,16 @@ def _set_csm_endpoint():
...
+            ep = Conf.get(const.CONSUMER_INDEX,
+                f'{const.RGW_S3_DATA_ENDPOINTS_KEY}[{ep_count}]')
+            if ep:
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:161]: You are raising an error just by checking the 0th index. What if your list is like - 
["", "Endpoint 1", "Endpoint 2"]

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/conf/configure.py (hunk 3 above)
Comment (rohitkolapkar) [reply csm/conf/configure.py:161]: @pawankumarsrivastava we are iterating over num_endpoints key present in confstore, whatever value is present there, these many valid values should be there in the list, that is why checking each value if not found then raising the error.

File: csm/conf/configure.py (hunk 2 above)
Comment (rohitkolapkar) [reply csm/conf/configure.py]: done

File: csm/conf/configure.py (hunk 3 above)
Comment (rohitkolapkar) [reply csm/conf/configure.py]: made changes as per our discussion. done.

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**